src/Persisters/Entity/BasicEntityPersister.php

```diff
@@ -737,7 +737,7 @@ public function load(
             $hints[Query::HINT_REFRESH_ENTITY] = $entity;
         }
 
-        $hydrator = $this->em->newHydrator($this->currentPersisterContext->selectJoinSql ? Query::HYDRATE_OBJECT : Query::HYDRATE_SIMPLEOBJECT);
+        $hydrator = $this->em->newHydrator($this->currentPersisterContext->selectJoinSql !== '' ? Query::HYDRATE_OBJECT : Query::HYDRATE_SIMPLEOBJECT);
```

Comment:
I will revert this change and the ones related to `selectJoinSql` below since they are inconsistent.

Reply:
The rest of the changes seem innocuous to me :thinking: